src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -120,6 +170,264 @@ public void recordTagValue(String time)
+      if (startTime.equalsIgnoreCase("-1")) {
+         Logger.LOG_ERR("EBD Error: Start time not set after reading time pointer file.");
+         return;
+      }
+
+      // end time should be the current time to always keep it up to date
+      endTime = convertTimeIntHelper(System.currentTimeMillis());
+
+      String tagName = getTagName();
+
+      try {
+          // https://ewonsupport.biz/ebd/ <- has a nice EBD string builder. :)
+          Exporter exporter = new Exporter("$dtHL$ftT$st"+startTime+"$et"+endTime+"$fnvalues$tn"+tagName);
+          exporter.ExportTo("file:///"+tagDataFileName);
+          exporter.close();
+      }catch (IOException ioe){
+          Logger.LOG_ERR(" error in export block descriptor");
+      }
+   }
+
+   /**
+    * parse a semi colon seperated file for the 'timeInt' and 'value' fields
+    */
+   public void parseTagFile()
+   {
+
+       BufferedReader reader;
+       try {
+           reader = new BufferedReader(new FileReader(tagDataFileName));
+           // read the first line and do nothing with it
+           String line = reader.readLine();
+           if (line != null)
+               line = reader.readLine(); // get first line of data
+
+           // counter to keep track of number of lines read in from historical logs
+           int count = 0;
+           while (line != null) {
+
+               // limit the lines read in to the max queue size to avoid massive queues
+               if(count > MAX_DATA_POINTS_PER_TAG) {
+                   // lines parsed has reached the max, stop parsing.
+                   break;
+               }
+               count++;
+
+               // extract the timeInt, which is first in the file
+               String timeInt = line.substring(0, line.indexOf(';'));
+               //Logger.LOG_INFO(timeInt);
+
+               // extract value, which is after 2 semi colons
+               String value = "";
+
+               /* this code finds the index of the substring where the tag's value is in the current line
+                * it looks at the number of occurrences of semi colons, and marks when the correct number has been
+                * seen. Then it records those indices to use in the substring call.
+                */
+               // main index for the loop
+               int indexMain = 0;
+               // recorded start index of new substring
+               int indexStart = 0;
+               // recorded end index of new substring
+               int indexEnd = 0;
+
+               int firstSemiColonOfInterest = 2;
+               int secondSemiColonOfInterest = 3;
+               int lastInterestingSemiColon = 4;
+               for(int i = 0; i < lastInterestingSemiColon; i++) {
+
+                   // need to add 1 to index main so indexOf function can work
+                   // or it will just find the same thing again
+                   indexMain = 1+ line.indexOf(';', indexMain);

```

Comment:
formatting is a bit strange, you have no space after the 1 then you have a space after the +

Reply:
I think they style i was formatting with put spaces around all +, <, ==, etc.

Ex: Logger.LOG_ERR("Tag \"" + tag.getTagName() + "\" does not exist on this eWON");